### DISCUSSION TITLE
next.config.js redirect improvements

### DISCUSSION THREAD

**Discussion by jserrao** [vercel/next.js#20405] [Ideas]
Next's handling for [redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) is already effective and clear. An improvement would be allowing `source` to accept arrays of values that can be reconciled to the same `destination`. This would reduce the total number of redirection blocks people would have to write (which can be intense on older, content-heavy sites). 

Proposed functionality:
```js
const nextConfig = {
  async redirects() {
    return [
      {
        source: [
          '/old-work',
          '/projects',
          '/case-studies'
        ],
        destination: '/work',
        permanent: true,
      }
    ]
  }
}
```

**Comment by mohsen1** [comment]
I can be wrong but since regular expressions are supported this can be done in one shot today too but I support this improvement because an array is more readable

**Comment by timneutkens** [comment]
The redirects configuration is a function that is executed so you can create your own handler for this if you want the feature.

**Reply by jserrao** [reply]
Interesting, thanks for the feedback.
